rules/common_settings.bzl

```diff
@@ -12,50 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Common build setting rules that return a BuildSettingInfo with the value of the build setting.
-For label-typed settings, use the native label_flag and label_setting rules."""
+"""Common build setting rules
 
-BuildSettingInfo = provider(fields = ["value"])
+These rules return a BuildSettingInfo with the value of the build setting.
+For label-typed settings, use the native label_flag and label_setting rules.
+
+More documentation on how to use build settings at
+https://docs.bazel.build/versions/master/skylark/config.html#user-defined-build-settings
+"""
+
+BuildSettingInfo = provider(
+    doc = """A singleton provider that contains the raw value of a build setting""",
```

Comment:
nit: Triple quotes unnecessary if it's only a single line :)  
(there are multiple such instances in this file)

Reply:
Done.